include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -159,51 +160,62 @@ class pairwise_alignment_algorithm : protected policies_t...
      * \param[in] sequence2 The second sequence to compute the alignment for.
      */
     template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
-    int32_t compute_matrix(sequence1_t & sequence1, sequence2_t & sequence2)
+    int32_t compute_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
     {
         // ---------------------------------------------------------------------
         // Initialisation phase: allocate memory and initialise first column.
         // ---------------------------------------------------------------------
 
         this->reset_optimum(); // Reset the tracker for the new alignment computation.
 
-        thread_local score_matrix_single_column<int32_t> local_matrix{};
-        local_matrix.reset_matrix(std::forward<sequence1_t>(sequence1), std::forward<sequence2_t>(sequence2));
-        auto matrix_iter = local_matrix.begin();
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
```

Comment:
So I read a bit about the `thread_local` keyword and I still have a question because I don't know how we parallelize the alignment algorithm yet (I cannot find the discussion on Gitter as you can't search through the chat history, unfortunately, because I know you explained it once already, sorry about that):

If I parallelize before calling a certain function, the local variables within that function are copies for each thread, right?
```cpp
foo()
{
    int i = 0;
    ++i;
    std::cout << i << std::endl;
}
// ...
#pragma omp parallel for
for (..)
    foo(); // will always print 1 because each thread has its own copy of i
```

Then why do you need the `thread_local` keyword here in the local scope of this function? I only found examples where a `thread_local` variable is at global scope.

Reply:
Because it will become a global variable for each thread which has the great advantage that the variable does not need to be constructed every time the function is called. The variable exists as long as the thread lives. If the construction is expensive like here where memory allocations happen then you safe the reallocations which would be otherwise needed.

Reply:
So it becomes static storage?

Reply:
Yes. If I read it correctly it is the same as `thread_local static` so the object has internal linkage and is bound to the lifetime of the thread.

Reply:
yeah even if they behave the same, I would also think that `static thread_local` would make it more obvious that this instance lives longer than the function invocation.